hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala

```diff
@@ -271,7 +271,7 @@ class HoodieMergeOnReadIncrementalHadoopFsRelationFactory(override val sqlContex
                                                           isBootstrap: Boolean)
   extends HoodieMergeOnReadSnapshotHadoopFsRelationFactory(sqlContext, metaClient, options, schemaSpec, isBootstrap) {
 
-  override val mandatoryFields: Seq[String] = Seq(HoodieRecord.COMMIT_TIME_METADATA_FIELD) ++ mandatoryFieldsForMerging
+  override val mandatoryFields: Seq[String] = Seq(HoodieRecord.COMMIT_TIME_METADATA_FIELD) ++ (if (isBootstrap) Seq.empty else partitionColumnsToRead.toSeq)
```

Comment:
You mentioned that "all this (mandatoryFieldsForMerging) is computed in the fg reader so we don't need this".  Do we need to move `mandatoryFields` logic to the file group reader too (OK to keep it in this PR; sth to consider for maintenability and cross-engine support, e.g., right now, reading partition values from data files is only supported in Spark?).

Reply:
Yes, this probably will be able to have cross engine support

Reply:
JIRA this?

Reply:
JIRA this?

Reply:
https://issues.apache.org/jira/browse/HUDI-8166